Dockerfile.pipelines

```diff
@@ -0,0 +1,117 @@
+FROM ubuntu:xenial
+
+# This DockerFile is looked after by
+MAINTAINER Tim Greaves <tim.greaves@imperial.ac.uk>
+
+# Add the ubuntu-toolchain-r test ppa
+RUN echo "deb http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu xenial main" > /etc/apt/sources.list.d/ubuntu-toolchain-r-ppa-xenial.list
+
+# Import the Launchpad PPA public key
+RUN gpg --keyserver keyserver.ubuntu.com --recv 1E9377A2BA9EF27F
+RUN gpg --export --armor BA9EF27F | apt-key add -
+
+# Ensure UTF-8 environment is correct for documentation builds
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+# Use bash, not dash, or later conditionals fail
+RUN echo "dash dash/sh boolean false" | debconf-set-selections
+RUN DEBIAN_FRONTEND=noninteractive dpkg-reconfigure dash
+
+# Upgrade to the most recent package set
+RUN apt-get update
+RUN apt-get -y dist-upgrade
+
+# Needed for the conda and devito installs later
+RUN apt-get -y install wget bzip2 git make
+
+# Default gcc version to install
+ARG gccVersion=8
+ENV DEVITO_ARCH=gcc-$gccVersion
+# Set Yask variables in case they are needed
+ENV YC_CXX=g++-$gccVersion
+
+# Default devito install method
+ARG installWithPip=false
+ENV testWithPip=$installWithPip
+
+# Use a different back-end?
+ARG DEVITO_BACKEND=none
+ENV DEVITO_BACKEND=$DEVITO_BACKEND
+
+# Use OpenMP?
+ARG DEVITO_OPENMP=0
+ENV DEVITO_OPENMP=$DEVITO_OPENMP
+ENV OMP_NUM_THREADS=2
+
+# Run examples?
+ARG RUN_EXAMPLES='false'
+ENV RUN_EXAMPLES=$RUN_EXAMPLES
+
+# Ensure a non-interactive matplotlib backend
+ENV MPLBACKEND=agg
```

Comment:
not sure if actually still any useful?

Reply:
Gerard commented that this would be a useful backstop in case we have problems along similar lines in future (this is far from the first time I've hit matplotlib backend issues...) and I'm inclined to agree with him unless there are concrete reasons to take it out :-)